### ISSUE TITLE
Update README

### ISSUE THREAD

**Issue by Diaoul** [SynoCommunity/spksrc#1727]
- [ ] Docker instructions
- [ ] Better explanations (shorter)
- [ ] Links to the wiki
- [ ] Section on package requests: https://github.com/SynoCommunity/spksrc/issues/1727#issuecomment-107898696

**Comment by Dr-Bean** [comment]
A possible addition to the list: 
- Section on package requests. Content: 
 - Before opening a new request, check open requests. 
 - Provide information in the request, e.g. project description, link to project homepage, documentation, license, source code. 
 - Don't ask for ETA's, do show support (`+1`, bounties or whatever's cool nowadays ;))

**Comment by cytec** [comment]
i've hacked together a quick draft for the request part:

Requests
------
Before opening a new Package request, check if there already is an open request for that Package in the [Issues Section](https://github.com/SynoCommunity/spksrc/labels/request). 

If you open a new Request there are some infromations that should be included with your Request, you may use this template:

Title:
```
[request] Name of Tool
```

Content:
```
Website: http://www.cool-website.com
Description: What does it?
Documentation: Link to documentation
Prerequisites: curl, openssl, etc...
Build method: any additional build information?
Sourcecode: url to the sourcecode
License: GPLv2, Apache, etc
```
Please don't ask for ETA's (it's done when it's done) but show your support (`+1`, bountysource, etc).

**Comment by Dr-Bean** [comment]
@cytec Nice work, that's exactly what I had in mind :) I've taken the opportunity to clarify some things directly in your comment.

As another addition to the list: I think a link to a general FAQ page would be helpful (user-focused, not developer-focused). 
I have collected a couple small things to put there (mostly one-liner type of things), but one thing that should end up there are permissions-related questions (especially when that one PR is merged). I don't have the text yet, but for now, a placeholder wiki article would be sufficient.

**Comment by Dr-Bean** [comment]
Add section on how to open an issue. Some points off the top of my head:
- Mention model and arch (link to our wiki)
- Mention DSM version
- Provide log files (most packages: check Package Center/Show log, or `/usr/local/{package}/var/`. Probably mention WinSCP/whatever the MAC equivalent)
- Run package via CLI: `/var/packages/{package}/scripts/start-stop-status start`, and provide output. Probably mention Putty (Called Terminal for MAC, iirc)
- Use gist.github.com/pastebin for logs over 10 lines.

**Comment by cytec** [comment]
@Dr-Bean @Diaoul how about adding some sort of "coding style guidelines"? 
Stuff like indent with spaces/tabs
New line at end of file etc?

**Comment by Diaoul** [comment]
I don't think we need to mention the obvious, like spaces/tabs/newlines.

* Use patches not replace (sed)
* Don't hardcode stuff, use variables

**Comment by Dr-Bean** [comment]
And linking back here too: https://github.com/SynoCommunity/spksrc/pull/2066. Take a look. I think we'd be best off by creating the various wiki articles first, then merge this

**Comment by Dr-Bean** [comment]
Added https://github.com/SynoCommunity/spksrc/wiki/Frequently-Asked-Questions (user-focused). I think I've covered the most obvious questions here, but feel free to append.

**Comment by cytec** [comment]
@SynoCommunity/developers how about a page describing the available options in makefiles? what you can set and when to use what... like `BETA` etc? currently i'm more or less copy/pasting stuff from other packages and search for the values all the time :D

**Comment by Dr-Bean** [comment]
Sure, go ahead. It's either that, or some kind of template package. I think I like the wiki better, especially to start off.
A while ago I was preparing some text for that purpose, I should see if I can dig it up.

**Comment by cytec** [comment]
well i've started a new page for that purpose here: https://github.com/SynoCommunity/spksrc/wiki/Makefile-variables not sure if i got everything covered and if it's 100% correct... just added the stuff i recall from memory for now

also i'm not sure how to handle variables which are not thought to be overwritten like `$(ARM_ARCHES)`, `$(RUN)`  etc

**Comment by Dr-Bean** [comment]
Here's the text I was working on...four days away from being exactly a year ago: https://gist.github.com/Dr-Bean/379f2421da7d9772b3cd. Might help a bit.

**Comment by Dr-Bean** [comment]
I believe this issue is mostly done, although of course there's always work to be done on improving wiki stuff.

@piwi82 made a good point in another issue about not using the current donation links in the README. 
So, unless someone speaks up ( @SynoCommunity/developers ), I'll replace them with the same as on the SynoCommunity website.
It directs general donations towards infra where it belongs, and anything else is up to the discretion of Diaoul. The rest is covered with BountySource.

**Comment by Dr-Bean** [comment]
Replaced the donation links with adf3867a997952a50673c13c07aed55f8ed6ad5f.

@cytec: the Makefile variables were added onto by @sirjaymz (thanks!), and I made a number of additional changes after that. The gist I linked to above might still come in handy in case you wanted to continue looking into adding onto the article.

Closing this one in the meantime, the original request has been fulfilled imo ;)
